internal/hmyapi/apiv1/blockchain.go

```diff
@@ -701,13 +767,57 @@ func (s *PublicBlockChainAPI) GetAllDelegationInformation(ctx context.Context, p
 
 // GetDelegationsByDelegator returns list of delegations for a delegator address.
 func (s *PublicBlockChainAPI) GetDelegationsByDelegator(ctx context.Context, address string) ([]*RPCDelegation, error) {
+	if err := s.isBeaconShard(); err != nil {
+		return nil, err
+	}
 	delegatorAddress := internal_common.ParseAddr(address)
 	validators, delegations := s.b.GetDelegationsByDelegator(delegatorAddress)
 	result := []*RPCDelegation{}
 	for i := range delegations {
 		delegation := delegations[i]
 
-		undelegations := []RPCUndelegation{}
+		undelegations := make([]RPCUndelegation, len(delegation.Undelegations))
+
+		for j := range delegation.Undelegations {
+			undelegations = append(undelegations, RPCUndelegation{
```

Comment:
`undelegations` is fixed-size array and does not need `append`. simply assign to index `j`. you can also do the same for `result`, as you already know the size, just use `make` and assign to result using index `I`. `append` is not recommended when your array size is known. you could do the same for `internal/hmyapi/apiv2/blockchain.go`

Reply:
also do the same for `GetDelegationsByDelegatorByBlockNumber`